08_multibundle_requirejs/Gruntfile.js

```diff
@@ -0,0 +1,286 @@
+
+                    // lib
+                    'app/lib/**/*.js',
+
+                    'app/views/home/**/*.js',
+                    'app/controllers/home_controller.js',
+                ],
+                // Creates separate bundle for user page components – `<destination>/user.<hash>.js`
+                user: [
+                    'app/models/users_bundle/**/*.js',
+                    'app/collections/users_bundle/**/*.js',
+                    'app/views/users_bundle/**/*.js',
+                    'app/controllers/users_bundle/**/*.js',
+                    'public/js/app/templates/users_bundle/compiledTemplates.js'
+                ],
+
+                // Creates separate bundle for repos page components – `<destination>/repos.<hash>.js`
+                repos: [
+                    'app/models/repos_bundle/**/*.js',
+                    'app/collections/repos_bundle/**/*.js',
+                    'app/views/repos_bundle/**/*.js',
+                    'app/controllers/repos_bundle/**/*.js',
+                    'public/js/app/templates/repos_bundle/compiledTemplates.js'
+                ]
+            }
+        },
+
+        handlebars: {
+            compile_server: {
+                options: {
+                    namespace: false,
+                    commonjs: true,
+                    processName: function(filename) {
+                        return filename.replace('app/templates/', '').replace('.hbs', '');
+                    }
+                },
+                src: 'app/templates/**/*.hbs',
+                dest: 'app/templates/compiledTemplates.js',
+                filter: function(filepath) {
+                    var filename = path.basename(filepath);
+                    // Exclude files that begin with '__' from being sent to the client,
+                    // i.e. __layout.hbs.
+                    return filename.slice(0, 2) !== '__';
+                }
+            },
+            /*
+             * Create separate compiled templates so that they can be bundled
+             */
+            compile_client: {
+                options: {
+                    amd: true,
+                    processName: function(filename) {
+                        return filename.replace('app/templates/', '').replace('.hbs', '');
+                    }
+                },
+                files: {
+                    'public/js/app/templates/compiledTemplates.js': ['app/templates/**/*.hbs',
+                        '!' + usersBundleTemplatePath, // Exclude since we'll build it separate
+                        '!' + reposBundleTemplatePath, // Exclude since we'll build it separate
+                        '!app/templates/__layout.hbs'
+                    ],
+                    'public/js/app/templates/users_bundle/compiledTemplates.js': usersBundleTemplatePath,
+                    'public/js/app/templates/repos_bundle/compiledTemplates.js': reposBundleTemplatePath
+                }
```

Comment:
This is something we haven't (fully) achieved either, and I "blame" Rendr original flow – having all the templates loaded at all times. What we have in Trulia now is just half-way to the real solution, pre compiled templates required with views. 
And for this example it might require different handlebars adapter. (Need to investigate).

Better way would be to require templates as is using requires plugins (`'hbs!path/to/template'`). This is next step for us in Trulia and probably it will propagate to this (or another) example.